Src/BootCamp.Chapter/BalanceStats.cs

```diff
@@ -7,31 +11,335 @@ public static class BalanceStats
+
+            for (int i = 0; i < peopleAndBalances.Length; i++)
+            {
+                string[] auxBalances = peopleAndBalances[i].Split(',');
+                decimal[] balances = GetBalances(auxBalances[1..]);
+                decimal currentMax = MaxBalance(balances);
+                if (max < currentMax )
+                {
+                    max = currentMax;
+                    peopleWithSameHighBalance.Clear();
+                    peopleWithSameHighBalance.Append(auxBalances[0]);
+                    moreThanOne = false;
+                    maxResult = $" had the most money ever. {max:C0}.";
+                }
+               else if (max.Equals(currentMax))
+                {
+                    peopleWithSameHighBalance.Append($", {auxBalances[0]}");
+                    moreThanOne = true;
+                }
+            }
+
+            if (moreThanOne)
+            {
+                ReplaceLastCommaWithAnd(peopleWithSameHighBalance);
+            }
+            peopleWithSameHighBalance.Append(maxResult);
+            return peopleWithSameHighBalance.ToString();
         }
 
         /// <summary>
         /// Return name and loss of a person with a biggest loss (balance change negative).
         /// </summary>
         public static string FindPersonWithBiggestLoss(string[] peopleAndBalances)
         {
-            return "";
+            if (IsStrArrayNullOrEmpty(peopleAndBalances))
+            {
+                return "N/A.";
+            }
+
+            CultureInfo.CurrentCulture = GetCustomInvariantCulture();
+
+            decimal min = decimal.MaxValue;
+            string minResult = "";
+            StringBuilder peopleWithSameLowBalance = new StringBuilder();
+            bool moreThanOne = false;
+
+            for (int i = 0; i < peopleAndBalances.Length; i++)
+            {
+                string[] auxBalances = peopleAndBalances[i].Split(',');
+                decimal[] balances = GetBalances(auxBalances[1..]);
+
+                if (balances.Length <= 1)
+                {
+                    return "N/A.";
+                }
+         
+                decimal currentMin = MinBalance(GetDiff(balances));
+                if (min > currentMin)
+                {
+                    min = currentMin;
+                    peopleWithSameLowBalance.Clear();
+                    peopleWithSameLowBalance.Append(auxBalances[0]);
+                    moreThanOne = false;
+                    minResult = $" lost the most money. {min:C0}.";
+                }
+                else if (min.Equals(currentMin))
+                {
+                    peopleWithSameLowBalance.Append($", {auxBalances[0]}");
+                    moreThanOne = true;
+                }
+            }
+
+            if (moreThanOne)
+            {
+                ReplaceLastCommaWithAnd(peopleWithSameLowBalance);
+                //int lastComma = peopleWithSameLowBalance.ToString().LastIndexOf(',');
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Remove this commented out code.](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=5069037)